test/src/test/java/org/corfudb/integration/LogReplicationAbstractIT.java

```diff
@@ -260,6 +270,53 @@ public void testEndToEndSnapshotAndLogEntrySyncUFO(int totalNumMaps, boolean dis
         }
     }
 
+    private void triggerSnapshotAndTestRemainingEntries() throws Exception{
+
+        // enforce a snapshot sync
+        Table<ExampleSchemas.ClusterUuidMsg, ExampleSchemas.ClusterUuidMsg, ExampleSchemas.ClusterUuidMsg> configTable =
+                corfuStoreActive.openTable(
+                        DefaultClusterManager.CONFIG_NAMESPACE, DefaultClusterManager.CONFIG_TABLE_NAME,
+                        ExampleSchemas.ClusterUuidMsg.class, ExampleSchemas.ClusterUuidMsg.class, ExampleSchemas.ClusterUuidMsg.class,
+                        TableOptions.fromProtoSchema(ExampleSchemas.ClusterUuidMsg.class)
+                );
+        try (TxnContext txn = corfuStoreActive.txn(DefaultClusterManager.CONFIG_NAMESPACE)) {
+            txn.putRecord(configTable, DefaultClusterManager.OP_ENFORCE_SNAPSHOT_FULL_SYNC,
+                    DefaultClusterManager.OP_ENFORCE_SNAPSHOT_FULL_SYNC, DefaultClusterManager.OP_ENFORCE_SNAPSHOT_FULL_SYNC);
+            txn.commit();
+        }
+
+        // Sleep, so we have remainingEntries populated from the scheduled polling task instead of
+        // from method that marks snapshot complete
+        TimeUnit.SECONDS.sleep(longInterval);
+
+        corfuStoreActive.openTable(LogReplicationMetadataManager.NAMESPACE,
+                LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        LogReplicationMetadata.ReplicationStatusKey key =
+                LogReplicationMetadata.ReplicationStatusKey
+                        .newBuilder()
+                        .setClusterId(DefaultClusterConfig.getStandbyClusterId())
+                        .build();
+        LogReplicationMetadata.ReplicationStatusVal replicationStatusVal;
+        while(true) {
```

Comment:
It's a better practice to use CountdownLatch and not burn CPU cycles in a loop waiting for a condition to be met. In this case if you have a listener on the status table you can easily decrease your latch on the expected conditions.

Reply:
makes sense. Will implement/use a listner.